nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/permissions/test/TestCommentManager.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * (C) Copyright 2006-2019 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nour AL KOTOB
+ */
+
+package org.nuxeo.ecm.platform.comment.permissions.test;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.core.api.CloseableCoreSession;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.security.ACE;
+import org.nuxeo.ecm.core.api.security.ACL;
+import org.nuxeo.ecm.core.api.security.ACP;
+import org.nuxeo.ecm.core.api.security.SecurityConstants;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.ecm.platform.comment.api.CommentManager;
+import org.nuxeo.ecm.platform.comment.service.CommentServiceHelper;
+import org.nuxeo.ecm.platform.comment.workflow.utils.CommentsConstants;
+import org.nuxeo.ecm.platform.usermanager.UserManager;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+@RunWith(FeaturesRunner.class)
+@Features({ CommentFeature.class })
+// @Deploy("org.nuxeo.ecm.platform.comment.tests:OSGI-INF/relation-comment-manager-contrib.xml")
+public class TestCommentManager {
+
+    public static final String QUERY_COMMENTS_AS_DOCUMENTS = "SELECT * FROM " + CommentsConstants.COMMENT_DOC_TYPE;
+
+    public static final String USERNAME = "Foo";
+
+    public static final String COMMENT_CONTENT_PROPERTY_XPATH = "comment:text";
+
+    public static final String COMMENT_CONTENT = "This is my comment";
+
+    @Inject
+    protected CoreSession systemSession;
+
+    @Inject
+    protected CoreFeature coreFeature;
+
+    @Inject
+    protected UserManager userManager;
+
+    @Test
+    public void testCreateReadDelete() throws Exception {
+        CommentManager commentManager = CommentServiceHelper.getCommentService().getCommentManager();
```

Comment:
Seems safer to create the `CommentManagerImpl` directly to make sure we test the deprecated service. `CommentService#getCommentManager` uses a `commentManager` field that may contain the other implementation.

```java
CommentManager commentManager = new CommentManagerImpl(CommentServiceHelper.getCommentService().getConfig());
```